### PR TITLE
Add context reading and writing functionality

### PR DESCRIPTION
**Pull request by gbarroutlook**
## Description

Added support for reading and writing in GraphDB for multiple file contexts

## Related Issues

Previous reading and writing operations did not support multiple file contexts

## Proposed Changes

Added the parameter context_id in the reading and writing operations so the specific graph for a specific file context can be queried in graphdb. Also enabled creation of the corresponding context graphs if they do not exist already

## Checklist

- [ ] I have tested these changes locally.
- [ ] I have reviewed the code changes.
- [ ] I have updated the documentation (if necessary).
- [ ] I have added appropriate unit tests (if applicable).

### CODE REVIEW

File: common/graph_db_client.py [hunk 1]
```diff
@@ -13,11 +13,12 @@
...
 
 class GraphDBClient:
-    def __init__(self,graph_db_params: GraphDBConfig|None):
+    def __init__(self,graph_db_params: GraphDBConfig|None,context_id: str|None):
+        self.username_prefix=graph_db_params.username
         self._client=client.Client(
             url=f"wss://{graph_db_params.account_name}.gremlin.cosmos.azure.com:443/",
             traversal_source="g",
-            username=graph_db_params.username,
+            username=self.username_prefix+"-contextid-"+context_id,
```
Comment (amritpalms) [inline common/graph_db_client.py:21]: why we have contextid in the username and we should incline it to support other types of auth.
in prod, infact, we won't be using username/password.

File: common/graph_db_client.py [hunk 2]
```diff
@@ -42,56 +43,76 @@ def result_to_df(self,result) -> pd.DataFrame:
         df = pd.DataFrame(json_data)
         return df
 
-    def query_vertices(self) -> pd.DataFrame:
+    def query_vertices(self,context_id:str="00000000-0000-0000-0000-000000000000") -> pd.DataFrame:
```
Comment (amritpalms) [inline common/graph_db_client.py:46]: if context-id is empty we should fail.

File: common/graph_db_client.py [hunk 3]
```diff
@@ -42,56 +43,76 @@ def result_to_df(self,result) -> pd.DataFrame:
...
         result = self._client.submit(
             message=(
                 "g.E()"
             ),
         )
         return self.result_to_df(result)
 
-    def write_vertices(self,data: pd.DataFrame)->None:
-        for row in data.itertuples():
-            print(row.id)
```
Comment (amritpalms) [inline common/graph_db_client.py:63]: is it intentional?

File: graphrag/index/cli.py [hunk 4]
```diff
@@ -108,7 +108,7 @@ def index_cli(
             ValueError("ContextOperation is invalid: It should be Active or DeActive")
         graphrag_config = _read_config_parameters(root, config, progress_reporter)
         _switch_context(graphrag_config, context_operation, context_id, progress_reporter)
-        sys.exit(0)
+        #sys.exit(0)
```
Comment (amritpalms) [inline graphrag/index/cli.py:111]: looks unintentional.

File: common/graph_db_client.py (hunk 1 above)
Comment (gbarroutlook) [reply common/graph_db_client.py:21]: This parameter is actually for the graph name, not really for authentication purposes. We can discuss more about what the naming format for the graphs should be

File: common/graph_db_client.py (hunk 3 above)
Comment (gbarroutlook) [reply common/graph_db_client.py:63]: This seems to be a result of git formatting the differences in a weird way. This function was not removed, just modified the parameters of the function

File: graphrag/index/cli.py (hunk 4 above)
Comment (gbarroutlook) [reply graphrag/index/cli.py:111]: I see, it was. I commented this to test the whole workflow of activating the context and indexing immediately after that. However, if the expected behavior is to do this in to separate steps, I will keep this line uncommented

**Pull request merged**